sdks/python/apache_beam/runners/runner.py

```diff
@@ -158,181 +177,39 @@ def run_pipeline(
     # type: (...) -> PipelineResult
 
     """Execute the entire pipeline or the sub-DAG reachable from a node.
-
-    Runners should override this method.
     """
-    raise NotImplementedError
+    pipeline.visit(
+        group_by_key_input_visitor(
+            not options.view_as(TypeOptions).allow_non_deterministic_key_coders)
+    )
+
+    # TODO: https://github.com/apache/beam/issues/19168
+    # portable runner specific default
```

Comment:
we can and plan to make this a  default for dataflow as well: https://github.com/apache/beam/issues/26996

Reply:
+1